src/expr/udf/src/external.rs

```diff
@@ -41,25 +46,40 @@ pub struct ArrowFlightUdfClient {
 impl ArrowFlightUdfClient {
     /// Connect to a UDF service.
     pub async fn connect(addr: &str) -> Result<Self> {
-        let conn = tonic::transport::Endpoint::new(addr.to_string())?
-            .timeout(Duration::from_secs(5))
-            .connect_timeout(Duration::from_secs(5))
-            .connect()
-            .await?;
-        let client = FlightServiceClient::new(conn);
-        Ok(Self {
-            client,
-            addr: addr.into(),
-        })
+        Self::connect_inner(
+            addr,
+            ResolutionStrategy::Eager {
+                timeout: TokioDuration::from_secs(5),
+            },
+        )
+        .await
     }
 
-    /// Connect to a UDF service lazily (i.e. only when the first request is sent).
+    /// Connect to a UDF service lazily (i.e. only fetch IP list from DNS when the first request is sent).
```

Comment:
There seems no need to update the doc. "connect" and "resolve DNS" happen at the same time in `ginepro`'s implementation.

Reply:
reverted the update.